src/components/apps/launch/ParamGroups.js

```diff
@@ -0,0 +1,375 @@
+                            break;
+                    }
+
+                    return <FastField key={param.id} {...fieldProps} />;
+                })}
+            </ExpansionPanelDetails>
+        </ExpansionPanel>
+    );
+}, messages);
+
+const ParamsReviewLabel = ({ error, label }) => {
+    return <Typography color={error ? "error" : "initial"}>{label}</Typography>;
+};
+
+const ParamsReviewValue = ({ param }) => {
+    const { value, type } = param;
+
+    switch (type) {
+        case constants.PARAM_TYPE.MULTILINE_TEXT:
+        case constants.PARAM_TYPE.MULTIFILE_SELECTOR:
+            return (
+                <TextField
+                    multiline
+                    rows={3}
+                    variant="outlined"
+                    margin="dense"
+                    fullWidth
+                    InputProps={{
+                        readOnly: true,
+                    }}
+                    value={Array.isArray(value) ? value.join("\n") : value}
+                />
+            );
+
+        case constants.PARAM_TYPE.TEXT_SELECTION:
+        case constants.PARAM_TYPE.INTEGER_SELECTION:
+        case constants.PARAM_TYPE.DOUBLE_SELECTION:
+            if (value?.display) {
+                return value.display;
+            }
+            break;
+
+        case constants.PARAM_TYPE.REFERENCE_GENOME:
+        case constants.PARAM_TYPE.REFERENCE_SEQUENCE:
+        case constants.PARAM_TYPE.REFERENCE_ANNOTATION:
+            if (value?.name) {
+                return value.name;
+            }
+            break;
+
+        default:
+            break;
+    }
+
+    return value;
+};
+
+/**
+ * A table summarizing the app parameter values and step resource requirements
+ * that will be included in the final analysis submission.
+ */
+const ParamsReview = injectIntl(({ appType, groups, errors, intl }) => (
+    <>
+        {appType === GlobalConstants.APP_TYPE_EXTERNAL && (
+            <Typography
+                variant="body1"
+                gutterBottom
+                dangerouslySetInnerHTML={{
+                    __html: formatMessage(intl, "hpcAppWaitTimes"),
+                }}
+            />
+        )}
+
+        <TableContainer component={Paper}>
+            <Table>
+                <TableBody>
+                    {groups?.map((group, groupIndex) =>
+                        group.parameters.map((param, paramIndex) => {
+                            const fieldName = `groups.${groupIndex}.parameters.${paramIndex}`;
+                            const paramError = getIn(errors, fieldName);
+                            const hasValue = !isEmptyParamValue(param.value);
+
+                            return (
+                                param.isVisible &&
+                                (paramError || hasValue) && (
+                                    <TableRow key={param.id}>
+                                        <TableCell>
+                                            <ParamsReviewLabel
```

Comment:
Parameters will only display in the `Review` step if they have a value set (either by default or by the user), or if they have a validation error.

Reply:
Hmm 🤔 What if they have accidentally left a field. I am leaning toward displaying all the parameters in the review step irrespective of user selection.

Reply:
In that case I don’t see too much of a difference between the param step and the review step. If there are only 1 or 2 required params, and 10 or 20 optional params, then I would think having 10+ blank fields in the review step would cause a lot of noise.

This reminds me to find an app in prod with a lot of optional params (I’ve seen at least one with close to 20 params spread in a bunch of groups before).

Reply:
The difference is you cannot edit values directly in the review step ;)

Exactly why I would like see all parameters in the review step. If there are 20 parameters, it is easy a miss one even though it is optional but needed for that analyses for that user.  Again these parameters will be grouped in the review step as well. So I don't see why we cannot include them here.

Reply:
Visual noise is something we need to understand from watching the user interact with this screen. If it is evident that users are distracted with this, I am in for removing / hiding these as an optional feature.

Reply:
I guess my thinking was that if the user wanted to review all the parameters, including the ones left blank, then the user could simply review the Params step, and there's no point duplicating that info in the Review step.

Although simply showing the param labels and values, without Info text and the groups between the params in the Review step, is a different view from the Params step 🤔

I agree that we should get input from the sci-team for this.

In the meantime, I found a public app (with 119 analysis completed so far) that has over 50 parameters that I can use in a story, and maybe provide screenshots of the alternate Review step views.

Reply:
I was also imagining an app that had a lot of optional parameters, but the user only needed to pick 1 or 2 out of the bunch, and in that scenario I thought only showing the parameters the user entered would be more convenient.

Using the actual app from prod with ~50 parameters, however, I see an example where most of those params are required, or have a default value already set. There are also a couple of pairs of params that are not marked as required, but the user must still provide a value for 1 of the params in the pair. In this scenario, I can see your point where they could easily overlook one of these not-required-but-still-required params if they're hidden from the Review step.

Reply:
Thanks @psarando for digging into this further. If It is not too much trouble, can you find out how many apps are in prod with 1. <= 10 params  2. 10 - 25 params  3.  > 25 before next UX meeting. And then we can discuss more about this in our upcoming UX meeting

Reply:
Again my comments on this issue are not show stopper. So feel free to merge anytime 👍